shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2716,19 +2726,35 @@ impl KafkaSinkCluster {
                     .unwrap()
                     .rack
                     .as_ref();
-                let shotover_node = self
-                    .shotover_nodes
-                    .iter()
-                    .find(|shotover_node| {
-                        coordinator_rack
-                            .map(|rack| rack == &shotover_node.rack)
-                            .unwrap_or(true)
-                    })
-                    .unwrap();
 
-                find_coordinator.host = shotover_node.address.host.clone();
-                find_coordinator.port = shotover_node.address.port;
-                find_coordinator.node_id = shotover_node.broker_id;
+                let hash = hash_str_bytes(request.key.clone());
+
+                let (shotover_nodes_in_rack, shotover_nodes_out_of_rack): (
+                    Vec<&ShotoverNode>,
+                    Vec<&ShotoverNode>,
+                ) = up_shotover_nodes.iter().partition(|shotover_node| {
+                    coordinator_rack
+                        .map(|rack| rack == &shotover_node.rack)
+                        .unwrap_or(true)
```

Comment:
Hey @rukai should it still be `unwrap_or(true)` here? i.e., if rack is none, should we consider shotover nodes in or out of the "rack"?

Reply:
yeah, `unwrap_or(true)` still makes sense.
If `coordinator_rack` is None then the kafka broker is configured without a rack.
So the in or out of rack distinction is meaningless and we can just pick any shotover broker to represent it.